### ISSUE TITLE
[feature request] restrict search to a certain field

### ISSUE THREAD

**Issue by tessus** [keepassxreboot/keepassxc#1461]
In certain situations searching all fields can yield too many results.

Here's an example:

You have an entry gmail with your user id and password. You search for gmail, because you are looking for your gmail account. In reality you get all accounts that use your gmail email address as the userid. This might be ok, if you have 10 entries in total, but if you have 200 entries and 80 of those are returned, it is a problem, because you still have to scroll through the result set to get to your gmail account.

Therefore it would be nice to be able to restrict the search to a single field. e.g. search in title only.

What do you think?

**Comment by seatedscribe** [comment]
What about adding a unique, characteristics word to that item? Open that
item and in the notes field write what you think will pop up in your mind
when doing the search. I do it for many of my accounts.

On 6 Feb 2018 05:41, "Helmut K. C. Tessarek" <notifications@github.com>
wrote:

In certain situations searching all fields can yield too many results.

Here's an example:

You have an entry gmail with your user id and password. You search for
gmail, because you are looking for your gmail account. In reality you get
all accounts that use your gmail email address as the userid. This might be
ok, if you have 10 entries in total, but if you have 200 entries and 80 of
those are returned, it is a problem, because you still have to scroll
through the result set to get to your gmail account.

Therefore it would be nice to be able to restrict the search to a single
field. e.g. search in title only.

What do you think?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/keepassxreboot/keepassxc/issues/1461>, or mute the
thread

**Comment by tessus** [comment]
@seatedscribe 

>What about adding a unique, characteristics word to that item?

You can't be serious. If I were able to remember 250 different unique words correlated to their respective accounts, I wouldn't need a password manager.

Also, I use the title to differentiate between the accounts. Call it a primary key, if you will.

It doesn't help though, if I can't search for the PK, but the entire data set.

**Comment by seatedscribe** [comment]
Well, maybe I misunderstood your words... Can you make a better example of
your use case and intent?

**Comment by tessus** [comment]
You are creating a workaround for an issue you have apparently experienced as well, which means it is a problem after all.

So why is it unfeasible to ask for a feature that solves that problem?

**Comment by seatedscribe** [comment]
Well, after all, I'm using a `certain field` so as to `restrict search`. You use the title, I use notes.

If a key is primary, how comes you get all the other items as well?

Your proposed solution is basically a list of fields, with just one to choose from and filter items based on that field, am I correct?

**Comment by tessus** [comment]
> If a key is primary, how comes you get all the other items as well?

Yep, exactly my point. I use it as a primary key, the app does not. I get all the other items, because the app is not just searching that field, but all fields.

The app allows you to create 2 entries with the same title, but that does not really matter, since I never do this anyway. All my title entries are unique. However, this doesn't help since the app searches all fields.

> Your proposed solution is basically a list of fields, with just one to choose from and filter items based on that field, am I correct?

This is one possible way to implement it. You could also provide the fields with check boxes and you check the ones you want to be searched. There are so many ways how you can implement this.
I don't really have a preference, as long as I can get it to search only in `title`.

Title is the account or the main defining field of an entry, thus I would just add an option 'search in title only' (next to `case sensitive` and `limit search to selected group`) . This would be the easiest and quickest way to implement it. Probably takes 10 minutes.
(I'm not a QT expert, thus it probably would take me much longer to browse through the code and put it together, but someone who's already familiar with the code could do this in less than 15 minutes.)

**Comment by seatedscribe** [comment]
> I use it as a primary key, the app does not.

 I am sorry if I still cannot get the meaning. If every title field is unique, as you said, you should be done with it.
Of course if you search for "gmail", I concur, you get rubbish.
What happens when you search for the {title} string instead? You will get just one (or a few) item. So I am a little confused...

**Comment by tessus** [comment]
> I am sorry I still cannot get the meaning. If every title field is unique, as you said, you should be done with it.

No, since I can't restrict the search just to that field.

> Of course if you search for "gmail", I concur, you get rubbish.

Right. I search for gmail which is the title for my gmail account. Yet I get all entries which have xxx@gmail.com as user id. But gmail is only one example. There are numerous. This is why I stated that in certain circumstances the search yields too many results.

> What happens when you search for the {title} string instead? You will get just one (or a few) item. So I am a little confused...

No, I don't. See my explanation above.

Unless you mean when the feature `search in title field only` will have been implemented.  In this case yes. And this is also the intent. When I search for gmail, I only want my gmail account, not 100 results where I use my gmail address as the user id.

**Comment by droidmonkey** [comment]
This is a cool debate, but ultimately this issue is a duplicate of #239.

**Comment by tessus** [comment]
@droidmonkey dude, you could've told me that BEFORE I wrote my fingers bloody.... ;-)

Cheers for the info !!! Will monitor #239 

(although #239 is far more complex than what I suggested.)

**Comment by droidmonkey** [comment]
Haha just checked my feed right before I posted the duplicate
